test/e2e/basic/csvs.go

```diff
@@ -0,0 +1,228 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/yaml"
+
+	olmoperators "github.com/operator-framework/api/pkg/operators/install"
+	olmv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+
+	ntoconfig "github.com/openshift/cluster-node-tuning-operator/pkg/config"
+	paocontroller "github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller"
+)
+
+var _ = ginkgo.Describe("[basic][clusterserviceversion] ClusterServiceVersion listing", func() {
+
+	Context("Saving current CSVs", func() {
+		var performanceOperatorCSVs []olmv1alpha1.ClusterServiceVersion
+		var cli client.Client
+		var existingCSVs int
+		var paginationLimit uint64
+
+		BeforeEach(func() {
+			var err error
+
+			By("Getting new client")
+			cli, err = newClient()
+			Expect(err).To(Succeed())
+
+			olmoperators.Install(cli.Scheme())
+
+			By("Listing all current CSVs")
+			performanceOperatorCSVs, err = getCurrentCSV(cli, context.TODO())
+			Expect(err).To(Succeed())
+
+			existingCSVs = len(performanceOperatorCSVs)
+
+			By(fmt.Sprintf("Delete all(%d) CSVs", existingCSVs))
+			for _, element := range performanceOperatorCSVs {
+				err = cli.Delete(context.TODO(), &element)
+				Expect(err).To(Succeed())
+			}
+
+			paginationLimit = uint64(1000)
+			By(fmt.Sprintf("Setting default paginationLimit to: %d", paginationLimit))
```

Comment:
deleting stuff makes me wary in general. Can we create the CSVs we need in out own testing namespace and leave the existing one alones?

Reply:
done